src/test/java/org/jenkinsci/plugins/argusnotifier/BuildMetricFactoryTest.groovy

```diff
@@ -43,4 +48,36 @@ class BuildMetricFactoryTest extends Specification {
         "myfolder"    | "myfolder." + FULL_JOB_NAME + ": " + BuildMetricFactory.NUMERIC_BUILD_STATUS_LABEL
         ""            | FULL_JOB_NAME + ": " + BuildMetricFactory.NUMERIC_BUILD_STATUS_LABEL
     }
+
+    def 'Total build time metric has git commit tag'() {
+        given:
+        long metricTimestamp = 42L
+        long totalDuration = 5L
+        int buildNumber = 349506
+        String scope = "testScope"
+        String jobFolderName = "a_jobFolderName"
+        folder.fullName >> jobFolderName
+        String commitTag = "GIT_COMMIT"
+        ImmutableMap.Builder<String, String> mapBuilder = ImmutableMap.<String, String>builder().put(commitTag,commitId);
+        Map<String, String> envVars = mapBuilder.build();
+        run.getEnvVars() >> envVars
+        run.getNumber() >> buildNumber
+
+        BuildMetricFactory metricFactory = new BuildMetricFactory(jenkins, run, metricTimestamp, scope)
+
+        when:
+        Metric metricTotalTime = metricFactory.getBuildTimeMetric(BuildMetricFactory.TOTAL_BUILD_TIME_LABEL,BuildMetricFactory.TOTAL_BUILD_TIME_METRIC,totalDuration)
+
+        then:
+
+        metricTotalTime.tags[commitTag.toLowerCase()] == expectedCommit
+        metricTotalTime.tags['build_number'] == commitId.isEmpty() ? null : String.valueOf(buildNumber)
```

Comment:
Is this really what we'd want? It seems like the absence of `GIT_COMMIT` shouldn't influence `BUILD_NUMBER`